util/git/git_test.go

```diff
@@ -458,48 +458,249 @@ func TestLsFiles(t *testing.T) {
-	require.NoError(t, err)
-	assert.Equal(t, expectedResult, lsResult)
+	require.NoError(t, os.Symlink(filepath.Join(tmpDir2, "c.yaml"), filepath.Join(tmpDir1, "link.yaml")))
 
-	// New and safer globbing, do not return symlinks resolving outside of the repo
-	expectedResult = []string{"a.yaml"}
-	lsResult, err = client.LsFiles("*.yaml", true)
+	require.NoError(t, runCmd(tmpDir1, "git", "add", "."))
+	require.NoError(t, runCmd(tmpDir1, "git", "commit", "-m", "Initial commit"))
+
+	tests := []struct {
+		name           string
+		pattern        string
+		safeGlobbing   bool
+		expectedResult []string
+	}{
+		{
+			name:           "Old globbing with symlinks and subdir",
+			pattern:        "*.yaml",
+			safeGlobbing:   false,
+			expectedResult: []string{"a.yaml", "link.yaml", "subdir/b.yaml"},
+		},
+		{
+			name:           "Safe globbing excludes symlinks",
+			pattern:        "*.yaml",
+			safeGlobbing:   true,
+			expectedResult: []string{"a.yaml"},
+		},
+		{
+			name:           "Safe globbing excludes external paths",
+			pattern:        filepath.Join(tmpDir2, "*.yaml"),
+			safeGlobbing:   true,
+			expectedResult: nil,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			lsResult, err := client.LsFiles(tt.pattern, tt.safeGlobbing)
+			require.NoError(t, err)
+			assert.Equal(t, tt.expectedResult, lsResult)
+		})
+	}
+}
+
+func TestLsFilesForGitFileGeneratorGlobbingPatterns(t *testing.T) {
+	tmpDir := t.TempDir()
+
+	client, err := NewClientExt("", tmpDir, NopCreds{}, false, false, "", "")
 	require.NoError(t, err)
-	assert.Equal(t, expectedResult, lsResult)
 
-	// New globbing, do not return files outside of the repo
-	var nilResult []string
-	lsResult, err = client.LsFiles(filepath.Join(tmpDir2, "*.yaml"), true)
+	err = runCmd(tmpDir, "git", "init")
 	require.NoError(t, err)
-	assert.Equal(t, nilResult, lsResult)
+
+	// Setup directory structure and files
+	setupDirs := []string{
+		"cluster-charts/cluster1/mychart/charts/mysubchart",
+		"cluster-charts/cluster1/myotherchart",
+		"cluster-charts/cluster2",
+		"some-path",
+		"some-path/staging",
+		"cluster-config/engineering/production",
+		"cluster-config/engineering/dev",
+		"p1/p2",
+		"p1/app2",
+		"p1/app3",
+	}
+	files := []string{
+		"cluster-charts/cluster1/mychart/charts/mysubchart/values.yaml",
+		"cluster-charts/cluster1/mychart/values.yaml",
+		"cluster-charts/cluster1/myotherchart/values.yaml",
+		"cluster-charts/cluster2/values.yaml",
+		"some-path/values.yaml",
+		"some-path/staging/values.yaml",
+		"cluster-config/engineering/production/config.json",
+		"cluster-config/engineering/dev/config.json",
+		"p1/p2/config.json",
+		"p1/app2/config.json",
+		"p1/app3/config.json",
+		"p1/config.json",
+	}
+
+	for _, dir := range setupDirs {
+		require.NoError(t, os.MkdirAll(filepath.Join(tmpDir, dir), 0o755))
+	}
+	for _, file := range files {
+		_, err := os.Create(filepath.Join(tmpDir, file))
+		require.NoError(t, err)
+	}
```

Comment:
instead of having a slice for directories and a slice for files, can we just have a slice for files and use `filepath.Dir` to get their directory and create them? 

```suggestion
	// Setup directory structure and files
	files := []string{
		"cluster-charts/cluster1/mychart/charts/mysubchart/values.yaml",
		"cluster-charts/cluster1/mychart/values.yaml",
		"cluster-charts/cluster1/myotherchart/values.yaml",
		"cluster-charts/cluster2/values.yaml",
		"some-path/values.yaml",
		"some-path/staging/values.yaml",
		"cluster-config/engineering/production/config.json",
		"cluster-config/engineering/dev/config.json",
		"p1/p2/config.json",
		"p1/app2/config.json",
		"p1/app3/config.json",
		"p1/config.json",
	}

	for _, file := range files {
		dir := filepath.Dir(file)
		require.NoError(t, os.MkdirAll(filepath.Join(tmpDir, dir), 0o755))
		_, err := os.Create(filepath.Join(tmpDir, file))
		require.NoError(t, err)
	}
```

Reply:
Good one, thank you for bringing it up. I've implemented the changes.